src/macos.rs

```diff
@@ -1,22 +1,17 @@
+use crate::Mode;
+use anyhow::Result;
 use std::process;
 
-fn is_dark_mode_enabled() -> bool {
-    if let Ok(output) = process::Command::new("defaults")
+pub fn detect() -> Result<Mode> {
```

Comment:
```suggestion
pub fn detect() -> Mode {
```

Changing the type of this return value is a breaking change to the public API. Does it help to return a Result here? What would an application do with an Err? Probably fall back to light mode, which this library does already,

Reply:
Done!